protobuf-delta-lake-sink/src/proto/parse.rs

```diff
@@ -0,0 +1,560 @@
+use std::{any::Any, sync::Arc};
+
+use deltalake::arrow::datatypes::{Field, Fields};
+use deltalake::{
+    arrow::{
+        array::{
+            Array, ArrayBuilder, ArrayData, ArrayRef, BinaryBuilder, BooleanBuilder, BufferBuilder,
+            Date32Array, Decimal128Array, GenericListArray, PrimitiveBuilder,
+            StringBuilder, StructArray,
+        },
+        datatypes::{ArrowPrimitiveType, DataType, Float32Type, Float64Type, Int32Type, Int64Type},
+        record_batch::RecordBatch,
+    },
+    Schema, SchemaTypeStruct,
+};
+use protobuf::reflect::{EnumDescriptor, ReflectValueBox};
+use protobuf::{
+    reflect::{FieldDescriptor, MessageDescriptor, ReflectValueRef, RuntimeType},
+    MessageDyn,
+};
+
+use super::get_delta_schema;
+
+trait ReflectBuilder: ArrayBuilder {
+    fn append_value(&mut self, v: Option<ReflectValueRef>) -> ();
```

Comment:
```suggestion
    fn append_value(&mut self, v: Option<ReflectValueRef>);
```

functions returning nil is the default, you don't need to specify it